src/parser.js

```diff
@@ -1535,6 +1535,32 @@ export class Parser extends Tokenizer {
           new Shift.BindingIdentifier(new Shift.Identifier(key.value)),
           this.parseAssignmentExpression()
         ), startTokenIndex);
+
+      } else if(this.match(TokenType.LPAREN)) {
+        // Parse the method parameters and the function body
+        let parmInfo = this.parseParams(null);
+        let [body, isStrict]= this.parseFunctionBody();
+
+        // Get all the local declarations from the function body
+        let decls = [];
+        body.statements.forEach(function(statement) {
```

Comment:
It's not just top-level `let`/`const` that we need to be worried about -- they can also be nested. I think you should just ignore this restriction for now. We will probably add a full (optional) second pass for these kinds of things later.